src/scripts/main.js

```diff
@@ -1,5 +1,26 @@
 'use strict';
 
-document.addEventListener('click', (e) => {
-  // write code here
+document.querySelector('.wall').addEventListener('click', (e) => {
+  const spider = document.querySelector('.spider');
+  const wall = e.currentTarget;
+
+  // Отримуємо координати стіни та розміри павука
+  const wallRect = wall.getBoundingClientRect();
+  const spiderWidth = spider.offsetWidth;
+  const spiderHeight = spider.offsetHeight;
+
+
+  const positionX = e.clientX - wallRect.left + window.pageXOffset;
+  const positionY = e.clientY - wallRect.top + window.pageYOffset;
+
+  // Обмежуємо координати, щоб павук не виходив за межі стіни
+  const maxX = wallRect.width - spiderWidth;
+  const maxY = wallRect.height - spiderHeight;
+
+  // Встановлюємо позицію павука, обмежуючи його межі
+  const finalX = Math.max(0, Math.min(positionX - spiderWidth / 2, maxX));
+  const finalY = Math.max(0, Math.min(positionY - spiderHeight / 2, maxY));
```

Comment:
The logic here correctly adjusts the spider's position to align its center with the click position and ensures it stays within the wall boundaries. However, ensure that the calculations for `finalX` and `finalY` are applied correctly when setting the spider's position.